package com.unlimitedstuffltd.mixin;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.inventory.MerchantResultSlot;
import net.minecraft.world.item.ItemStack;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(MerchantResultSlot.class)
public class ScammerMixin {
    @Inject(method = "onTake", at = @At(value = "TAIL"))
    public void onTake(Player player, ItemStack carried, CallbackInfo ci) {
        player.containerMenu.setCarried(ItemStack.EMPTY);
        player.sendOverlayMessage(Component.literal("YOU GOT SCAMMED!"));
    }
}